api/proto/teleport/notifications/v1/notifications.proto

```diff
@@ -42,8 +42,8 @@ message Notification {
 
 // NotificationSpec is the notification specification.
 message NotificationSpec {
-  // id is the UUID of the notification, in UUIDv7 format.
-  string id = 1;
+  reserved 1;
+  reserved "id";
```

Comment:
Because we already have the notification ID in `Metadata.Name`, and there's no benefit of having it in two places, it could just lead to confusion and inconsistency in how we extract it. `Metadata.Name` is preferable over `Spec.Id` because it's consistent with where we store ID's for other resources, and also because the generic backend service wrapper expects and uses that field to determine the key under which to store the resource in the backend.